### ISSUE TITLE
cannot start jekyll in vagrant

### ISSUE THREAD

**Issue by csomor** [wxWidgets/website#55]
Hi

sorry, I'm really not fluent in vagrant, and just trying to create a simple PR for the website :-) but I stumble over 

```
vagrant@ubuntu-bionic:/vagrant$ jekyll serve --host 0.0.0.0
Traceback (most recent call last):
	5: from /usr/local/bin/jekyll:23:in `<main>'
	4: from /usr/local/bin/jekyll:23:in `load'
	3: from /var/lib/gems/2.5.0/gems/jekyll-3.8.5/exe/jekyll:11:in `<top (required)>'
	2: from /var/lib/gems/2.5.0/gems/jekyll-3.8.5/lib/jekyll/plugin_manager.rb:48:in `require_from_bundler'
	1: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
/usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require': cannot load such file -- bundler (LoadError)

```

having 

```
vagrant@ubuntu-bionic:/vagrant$ ruby --version
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux-gnu]

```

Any pointers are very appreciated. Thanks

**Comment by csomor** [comment]
ok, after fighting two hours with mismatching dependencies and trying to resolve things by hand, I finally found a radical, but usable workaround, removing all gems, adding the needed bundler and install from there ..

```
cd /vagrant
sudo gem uninstall -aIx
sudo gem i bundler:2.1.4
bundle install

```

**Comment by vadz** [comment]
Sorry, I have no idea about Vagrant as I've never used it myself, but just installing the required gems in my own path on a Linux (and I would imagine it would work the same under macOS) system works fine for me, i.e. I do something like
```sh
$ export GEM_PATH=~/ruby
$ gem i --user-install jekyll
$ ~/ruby/bin/jekyll serve ...
```

**Comment by csomor** [comment]
@vadz thanks a lot, I found out a workaround, so I can go on with my PR :-)

**Comment by doscott** [comment]
I went through several weeks a while back when VirtualBox would not run on Tumbleweed so I switched over to KVM. To get around Vagrant I used a jekyll docker container. I have two scripts:

jekyllbuild.sh:
-----------------
cd /home/dos/website

export JEKYLL_VERSION=3.8
docker run --rm -it \
  --name=jekyll \
  --volume="$PWD:/srv/jekyll" \
  --volume="$PWD/vendor/bundle:/usr/local/bundle" \
  -p 4000:4000 jekyll/jekyll:3.8 \
  jekyll build

jekyll.sh
-----------
cd /home/dos/website

export JEKYLL_VERSION=3.8
docker run --rm -it \
  --name=jekyll \
  --volume="$PWD:/srv/jekyll" \
  --volume="$PWD/vendor/bundle:/usr/local/bundle" \
  -p 4000:4000 jekyll/jekyll:3.8 \
  jekyll serve --watch

**Comment by csomor** [comment]
@doscott thanks a lot, Danny :-)

**Comment by tierra** [comment]
Sorry, I must have broke this when creating the GitHub Actions publish workflows a while back.

Vagrant is really mostly meant to be used in Windows. It can be used elsewhere, but on Linux and OSX, it's really better to just run Node.js, Ruby, and Jekyll in the OS natively, where it's not only fastest, but also better supports filesystem watchers.

How you install and use Node.js and Ruby is really varied between developers though, so the docs don't attempt to assume how anyone has their environments setup for those, and only outlines the steps to use Vagrant.

Anyway, I believe I've fixed the Vagrant environment in #57, please give that look.

**Comment by csomor** [comment]
@tierra Thanks Bryan, I'll test that
